examples/golang/simpleinterconnectprovider/handler/handler.go

```diff
@@ -99,39 +168,227 @@ func (p *PrototypeHandler) DeleteTransceiver(_ context.Context, trans *pb.Delete
 
 func (p *PrototypeHandler) ListContactWindows(context.Context, *pb.ListContactWindowsRequest) (*pb.ListContactWindowsResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method ListContactWindows not implemented")
+	// TODO: Handle filter
+	return &pb.ListContactWindowsResponse{
+		ContactWindows: p.contactWindows,
+	}, nil
 }
+
 func (p *PrototypeHandler) ListBearers(context.Context, *pb.ListBearersRequest) (*pb.ListBearersResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method ListBearers not implemented")
+	// TODO: Handle filter
+	bearers := make([]*pb.Bearer, 0, len(p.attachmentCircuits))
+	for _, bearer := range p.bearers {
+		bearers = append(bearers, bearer)
+	}
+
+	return &pb.ListBearersResponse{Bearers: bearers}, nil
 }
-func (p *PrototypeHandler) CreateBearer(context.Context, *pb.CreateBearerRequest) (*pb.Bearer, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method CreateBearer not implemented")
+
+func (p *PrototypeHandler) GetBearer(_ context.Context, bearer *pb.GetBearerRequest) (*pb.Bearer, error) {
+	if p.bearers[bearer.Name] == nil {
+		return nil, status.Errorf(codes.NotFound, "bearer with requested ID was not found")
+	}
+
+	return p.bearers[bearer.Name], nil
 }
-func (p *PrototypeHandler) DeleteBearer(context.Context, *pb.DeleteBearerRequest) (*emptypb.Empty, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method DeleteBearer not implemented")
+
+func (p *PrototypeHandler) CreateBearer(_ context.Context, bearer *pb.CreateBearerRequest) (*pb.Bearer, error) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	bearerName := fmt.Sprintf("bearers/%s", bearer.BearerId)
+	if p.bearers[bearerName] != nil {
+		return nil, status.Errorf(codes.AlreadyExists, "bearer with requested ID was already created")
+	}
+	if bearer.Bearer.Interval.StartTime.AsTime().After(bearer.Bearer.Interval.EndTime.AsTime()) {
+		return nil, status.Errorf(codes.InvalidArgument, "bearer has negative time interval argument")
+	}
+	if !p.checkForSufficientContactWindow(bearer.Bearer) {
+		return nil, status.Errorf(codes.FailedPrecondition, "bearer has no sufficient contact window")
+	}
+	// Override the name of the attachment circuit to ensure that it has the correct resource name.
+	// It is up to the API to either validate the correctness of the name or just override it on creation.
+	bearer.Bearer.Name = bearerName
+
+	p.bearers[bearerName] = bearer.Bearer
+
+	return bearer.Bearer, nil
 }
+
+func (p *PrototypeHandler) checkForSufficientContactWindow(bearer *pb.Bearer) bool {
+	for _, contactWindow := range p.contactWindows {
+		if contactWindow.Target != bearer.Target || contactWindow.Transceiver != bearer.Transceiver {
+			continue
+		}
+
+		if contactWindow.Interval.StartTime.AsTime().After(bearer.Interval.StartTime.AsTime()) ||
+			contactWindow.Interval.EndTime.AsTime().Before(bearer.Interval.EndTime.AsTime()) {
+			continue
+		}
+
+		if contactWindow.MinRxCenterFrequencyHz >= bearer.RxCenterFrequencyHz ||
```

Comment:
nit, I believe this logic as declared doesn't allow for equality (e.g. bearer's RxCenterFrequencyHz cannot be equal to contactWindow.MinRxCenterFrequencyHz?).  That seems like a minor typo (I would think it's OK if they are equal)?

Reply:
makes sense. I changed it and added a test for one of the cases.